### PR TITLE
Refactor: Open Previewer from CardBrowser using new Activity APIs

### PR DESCRIPTION
**Pull request by nicdard**
## Purpose / Description
The old method of passing information to / starting / receiving results from one Activity to/from another Activity is deprecated in general, here we start refactoring the code to use the new androidx.activity APIs for such communication

## Fixes
Fixes (only partially) https://github.com/ankidroid/Anki-Android/issues/8602

## Approach
Each `startActivityForResult` and corresponding case in `onActivityResult` is refactored to use the new API.
This first PR refactors only the communication between `Previewer` and `CardBrowser` in order to be sure that the style looks good before moving to the others.

## How Has This Been Tested?
On emulators: API 30 and API 25.

## Learning (optional, can help others)
https://developer.android.com/training/basics/intents/result

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java [hunk 1]
```diff
@@ -170,7 +172,25 @@
 
     private static final int EDIT_CARD = 0;
     private static final int ADD_NOTE = 1;
-    private static final int PREVIEW_CARDS = 2;
+
+    ActivityResultLauncher<Intent> mOnPreviewCardsActivityResult = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+        if (result.getResultCode() == DeckPicker.RESULT_DB_ERROR) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:177]: Logging here would be useful

**Review by david-allison-1 — COMMENTED** [review]
One quick change and good to go!

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java (hunk 1 above)
Comment (nicdard) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:178]: :+1:

**Review by david-allison-1 — APPROVED** [review]
LGTM

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java [hunk 2]
```diff
@@ -170,7 +172,26 @@
...
+            closeCardBrowser(DeckPicker.RESULT_DB_ERROR);
+        }
+        // Previewing can now perform an "edit", so it can pass on a reloadRequired
+        Intent data = result.getData();
+        boolean hasChanges = data != null && (
+                data.getBooleanExtra("reloadRequired", false)
+                        || data.getBooleanExtra("noteChanged", false)
+        );
+        if (hasChanges) {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:187]: I am really picky about compound conditionals, I like them to be as easy to read as possible - something more like this?

```suggestion
        if ((data != nul)l && 
            (data.getBooleanExtra("reloadRequired", false) || data.getBooleanExtra("noteChanged", false)) {
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
This LGTM with one nitpick on the compound conditional, all the logic seems great

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java (hunk 2 above)
Comment (nicdard) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:187]: I liked to have it in a separate help variable to allow the reader not to read the condition to understand the meaning of the if statement, but I can change it :muscle: :)

Comment (nicdard) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java]: Fixed!

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java]: I just prefer to go the other direction - make the conditional so easy to mentally parse that it is not a problem (where possible)

**Review by mikehardy — APPROVED** [review]
LGTM - thanks!

**Comment by nicdard** [comment]
@mikehardy One side note: I have seen that in `AnkiActivity` there are some help methods to start an activity with or without animation, this behavior is difficult to reproduce with the new API without replicating the same code for each call, do you have any tip about this issue?
For more clarity I can open a PR where I extract one of those calls, to let you have an example of the issue

**Comment by mikehardy** [comment]
@nicdard 

> @mikehardy One side note: I have seen that in `AnkiActivity` there are some help methods to start an activity with or without animation, this behavior is difficult to reproduce with the new API without replicating the same code for each call, do you have any tip about this issue?
> For more clarity I can open a PR where I extract one of those calls, to let you have an example of the issue

Yeah I think I'd have to see it, but perhaps there is some way to use inheritance, I'm really vague on this but maybe the implementers of the activity result contract can extend some super object that handles the animation or not

**Pull request merged**